File: two-sum/solutions/go/isaiah.go
package algo

// https://leetcode.com/submissions/detail/788059137/
func twoSum(nums []int, target int) []int {
	var output []int
	var bucket = make(map[int]int)

	for i := 0; i < len(nums); i++ {
		diff := target - nums[i]
		if j, s := bucket[diff]; s {
			return []int{j, i}
		} else {
			bucket[nums[i]] = i
		}
	}

	return output
}

File: two-sum/solutions/js/isaiah.js

// https://leetcode.com/submissions/detail/731631281/
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let rst = []
    let ctn = {}
    for(let i = 0; i < nums.length; i++) {
        let n = ctn[target - nums[i]]
        if(n >= 0) {
            rst.push(n, i)
            break
        }
        
        ctn[nums[i]] = i
    }
    return rst
};